src/components/ContentSidebar/ActivitySidebar.js

```diff
@@ -17,73 +24,435 @@ type ExternalProps = {
+        }
+    }
+
+    /**
+     * Success callback for fetching feed items
+     */
+    feedSuccessCallback = (): void => {
+        this.fetchFeedItems();
+    };
+
+    /**
+     * Error callback for fetching feed items
+     */
+    feedErrorCallback = (e: $AxiosXHR<any>) => {
+        this.errorCallback(e);
+        this.fetchFeedItems();
+    };
+
+    /**
+     * Deletes a task via the API.
+     *
+     * @param {Object} args - A subset of the task
+     */
+    deleteTask = ({ id }: { id: string }): void => {
+        const { file, api, onTaskDelete = noop } = this.props;
+        api.getFeedAPI(false).deleteTask(
+            file,
+            id,
+            (taskId: string) => {
+                this.feedSuccessCallback();
+                onTaskDelete(taskId);
+            },
+            this.feedErrorCallback
+        );
+
+        // need to load the pending item
+        this.fetchFeedItems();
+    };
+
+    /**
+     * Updates a task in the state via the API.
+     *
+     * @param {Object} args - A subset of the task
+     * @return void
+     */
+    updateTask = ({ text, id }: { text: string, id: string }): void => {
+        const { file, api } = this.props;
+        api.getFeedAPI(false).updateTask(file, id, text, this.feedSuccessCallback, this.feedErrorCallback);
+
+        // need to load the pending item
+        this.fetchFeedItems();
+    };
+
+    /**
+     * Updates the task assignment
+     *
+     * @param {string} taskId - ID of task to be updated
+     * @param {string} taskAssignmentId - Task assignment ID
+     * @param {string} status - New task assignment status
+     * @param {string} message - the message to the assignee
+     * @return void
+     */
+    updateTaskAssignment = (taskId: string, taskAssignmentId: string, status: string, message?: string): void => {
+        const { file, api } = this.props;
+
+        api
+            .getFeedAPI(false)
+            .updateTaskAssignment(
+                file,
+                taskId,
+                taskAssignmentId,
+                status,
+                message,
+                this.feedSuccessCallback,
+                this.feedErrorCallback
+            );
+
+        // need to load the pending item
+        this.fetchFeedItems();
+    };
+
+    /**
+     * Deletes a comment via the API.
+     *
+     * @param {Object} args - A subset of the comment
+     * @return void
+     */
+    deleteComment = ({ id, permissions }: { id: string, permissions: BoxItemPermission }): void => {
+        const { file, api, onCommentDelete = noop } = this.props;
+
+        api.getFeedAPI(false).deleteComment(
+            file,
+            id,
+            permissions,
+            (comment: Comment) => {
+                this.feedSuccessCallback();
+                onCommentDelete(comment);
+            },
+            this.feedErrorCallback
+        );
+
+        // need to load the pending item
+        this.fetchFeedItems();
```

Comment:
Since `feedSuccessCallback()` and `feedErrorCallback()` call `fetchFeedItems()` is this necessary? Unless the comment explains that... :)

Reply:
Those are called after the call is completed. We need to fetch the feed items immediately after the API is invoked, since the cache is updated with the pending item and we want to update the UI